sdk/servicebus/service-bus/test/serviceBusClient.spec.ts

```diff
@@ -311,79 +299,101 @@ describe("Errors with non existing Queue/Topic/Subscription", async function():
-//     sbClient = ServiceBusClient.createFromAadTokenCredentials(123 as any, tokenCreds);
-//     should.equal(sbClient.name, "sb://123/", "Name of the namespace is different than expected");
-//   });
-
-//   it("sends a message to the ServiceBus entity", async function(): Promise<void> {
-//     const tokenCreds = await getTokenCredentialsFromAAD();
-//     await testCreateFromAadTokenCredentials(serviceBusEndpoint, tokenCreds);
-//     await sbClient.close();
-//   });
-// });
+describe("Test ServiceBusClient creation #RunInBrowser", function(): void {
+  let sbClient: ServiceBusClient;
+  let errorWasThrown: boolean = false;
+
+  const env = getEnvVars();
+  const serviceBusEndpoint = (env.SERVICEBUS_CONNECTION_STRING.match(
+    "Endpoint=sb://((.*).servicebus.windows.net)"
+  ) || "")[1];
+
+  /**
+   * Utility to create EnvironmentCredential using `@azure/identity`
+   */
+  function getDefaultTokenCredential() {
+    should.exist(
+      env[EnvVarNames.AZURE_CLIENT_ID],
+      "define AZURE_CLIENT_ID in your environment before running integration tests."
+    );
+    should.exist(
+      env[EnvVarNames.AZURE_TENANT_ID],
+      "define AZURE_TENANT_ID in your environment before running integration tests."
+    );
+    should.exist(
+      env[EnvVarNames.AZURE_CLIENT_SECRET],
+      "define AZURE_CLIENT_SECRET in your environment before running integration tests."
+    );
+    should.exist(
+      env[EnvVarNames.SERVICEBUS_CONNECTION_STRING],
+      "define EVENTHUB_CONNECTION_STRING in your environment before running integration tests."
+    );
+    return new EnvironmentCredential();
+  }
+
+  it("throws error for invalid tokenCredentials", async function(): Promise<void> {
+    try {
+      new ServiceBusClient(serviceBusEndpoint, [] as any);
+    } catch (err) {
+      errorWasThrown = true;
+      should.equal(
+        err.message,
+        "Connection string malformed: each part of the connection string must have an `=` assignment.",
+        // "'credentials' is a required parameter and must be an implementation of TokenCredential when using host based constructor overload.",
+        "ErrorMessage is different than expected"
+      );
+    }
+    should.equal(errorWasThrown, true, "Error thrown flag must be true");
+  });
+
+  it("throws error for undefined tokenCredentials", async function(): Promise<void> {
+    try {
+      new ServiceBusClient(serviceBusEndpoint, undefined as any);
+    } catch (err) {
+      errorWasThrown = true;
+      should.equal(
+        err.message,
+        "Connection string malformed: each part of the connection string must have an `=` assignment.",
+        // "'credentials' is a required parameter and must be an implementation of TokenCredential when using host based constructor overload.",
+        "ErrorMessage is different than expected"
+      );
+    }
+    should.equal(errorWasThrown, true, "Error thrown flag must be true");
+  });
+
+  if (isNode) {
```

Comment:
What is the reason for the `isNode()` check here?

Reply:
That's because EnvironmentCredential is not supported in browser, so only a select subset of tests can be run in browser mode.

Carrying forward Q I had from other somewhat related comment - is `SharedKeyCredential` not going to be used and be removed from `core-amqp`? If not, was it supposed to work in browser?

Reply:
Regarding the question on `SharedKeyCredential`, see https://github.com/Azure/azure-sdk-for-js/pull/6770#discussion_r366530508

Looks like among all the credentials from azure/identity, only the interactive credential works in browser which of course we can't use in CI.

There is ongoing work to change that. For now, I believe all libraries using azure/identity are falling under this case where the credentials from the azure/identity library are not testable in browser in CI.